ContextDesktop/src/main/java/com/github/migi_1/Context/model/MainEnvironment.java

```diff
@@ -189,14 +192,14 @@ private void initSpatials() {
         platform = new Platform(PLATFORM_LOCATION);
         commander = new Commander(COMMANDER_LOCATION);
         carriers = createCarriers();
-        damageDealerGenerator = new DamageDealerGenerator(commander);
+        obstacleSpawner = new ObstacleSpawner(commander);
+
         //attach all objects to the root pane
         for (LevelPiece levelPiece : levelGenerator.getLevelPieces(COMMANDER_LOCATION)) {
             addDisplayable(levelPiece);
         }
-
-        for (DamageDealer damageDealer : damageDealerGenerator.getObstacles()) {
-            addDisplayable(damageDealer);
+        for (Obstacle staticObstacle : obstacleSpawner.getObstacles()) {
+            addDisplayable(staticObstacle);
         }
         for (Path path : levelGenerator.getPathPieces(COMMANDER_LOCATION)) {
             addDisplayable(path);
```

Comment:
Maybe put these four for-loops in a different method (like an addDisplayables method).